ddl/partition.go

```diff
@@ -116,9 +116,17 @@ func buildTablePartitionInfo(ctx sessionctx.Context, d *ddl, s *ast.CreateTableS
 }
 
 func buildHashPartitionDefinitions(ctx sessionctx.Context, d *ddl, s *ast.CreateTableStmt, pi *model.PartitionInfo) error {
-	genIDs, err := d.genGlobalIDs(int(pi.Num))
-	if err != nil {
-		return errors.Trace(err)
+	// When this function is called by MockTableInfo, we should set particular partition ids, all default 0 here.
```

Comment:
The code here is confusing. Maybe it's better to pass a non-nil `ddl` here, and then try to override the geneated partition ID.

Reply:
OK, addressed.